utils/oc.py

```diff
@@ -15,12 +16,30 @@ class JSONParsingError(Exception):
 
 
 class OC(object):
-    def __init__(self, server, token):
-        self.oc_base_cmd = ['oc', '--server', server, '--token', token]
+    def __init__(self, server, token, jh_data=None):
+        ssh_base_cmd = self.init_ssh_base_cmd(jh_data)
+        self.oc_base_cmd = ssh_base_cmd + \
+            ['oc', '--server', server, '--token', token]
 
     def whoami(self):
         return self._run(['whoami'])
 
+    def init_ssh_base_cmd(self, jh_data):
+        if jh_data is None:
+            return []
+
+        import tempfile
+
+        hostname = jh_data['hostname']
+        identity = jh_data['identity']
+        user = jh_data['user']
+        identity_file = tempfile.mkdtemp() + '/id'
+        with open(identity_file, 'w') as f:
+            f.write(identity)
+        s.chmod(identity_file, 0o600)
```

Comment:
this looks weird

Reply:
this is octal representation

Reply:
wasn't aware that 0600 will not work with Py3, looks good then.